File: browser/src/js/moneysocket/message/notification/provider_not_ready.js
// Copyright (c) 2020 Jarret Dyrbye
// Distributed under the MIT software license, see the accompanying
// file LICENSE or http://www.opensource.org/licenses/mit-license.php


const BinUtl = require('../../utl/bin.js').BinUtl;
const MoneysocketNotification = require(
    './notification.js').MoneysocketNotification;
let NOTIFICATION_SUBCLASSES = require(
    './notification.js').NOTIFICATION_SUBCLASSES;

class NotifyProviderNotReady extends MoneysocketNotification {
    constructor(request_reference_uuid) {
        super("NOTIFY_PROVIDER_NOT_READY", request_reference_uuid);
    }

    cryptLevel() {
        return "AES";
    }

    static castClass(msg_dict) {
        var c = new NotifyProviderNotReady(msg_dict['request_reference_id']);
        Object.keys(msg_dict).forEach(key => {
            c[key] = msg_dict[key];
        });
        return c;
    }

    static checkValidMsgDict(msg_dict) {
        return null;
    }
}

NOTIFICATION_SUBCLASSES['NOTIFY_PROVIDER_NOT_READY'] = (
    NotifyProviderNotReady);

exports.NotifyProviderNotReady = NotifyProviderNotReady;

File: python/moneysocket/message/notification/provider_not_ready.py
# Copyright (c) 2020 Jarret Dyrbye
# Distributed under the MIT software license, see the accompanying
# file LICENSE or http://www.opensource.org/licenses/mit-license.php

import string
import json

from moneysocket.message.notification.notification import (
    MoneysocketNotification)

class NotifyProviderNotReady(MoneysocketNotification):
    MUST_BE_CLEARTEXT = False

    def __init__(self, request_reference_uuid):
        super().__init__("NOTIFY_PROVIDER_NOT_READY",
                         request_reference_uuid=request_reference_uuid)

    @staticmethod
    def cast_class(msg_dict):
        c = NotifyProviderNotReady(msg_dict['request_reference_uuid'])
        c.update(msg_dict)
        return c

    @staticmethod
    def check_valid_msg_dict(msg_dict):
        return None


MoneysocketNotification.NOTIFICATION_SUBCLASSES[
    'NOTIFY_PROVIDER_NOT_READY'] = NotifyProviderNotReady
